python/lsst/ts/externalscripts/maintel/serpent_walk.py

```diff
@@ -0,0 +1,339 @@
+                if comp not in self.tcs.components_attr:
+                    self.log.warning(
+                        f"Component {comp} not in CSC Group. "
+                        f"Must be one of {self.tcs.components_attr}. Ignoring."
+                    )
+                else:
+                    self.log.debug(f"Ignoring component {comp}.")
+                    setattr(self.tcs.check, comp, False)
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script."""
+        metadata.duration = self.config.total_time
+
+    async def run(self):
+
+        azel_generator = self.azel_grid_by_time(
+            total_time=self.config.total_time,
+            az_grid=self.config.az_grid,
+            el_grid=self.config.el_grid,
+            el_cutoff=self.config.el_cutoff,
+        )
+
+        for i, az, el in azel_generator:
+            self.log.debug(f"Serpent Walk - Target {i} - Moving to {az} / {el}")
+            await self.slew_and_track(az, el, target_name=f"serpent_walk_{i}")
+
+    def azel_grid_by_time(self, total_time, az_grid, el_grid, el_cutoff):
+        """
+        Generate Az/El coordinates for a a long time so we can slew and track
+        to these targets using a predefined az_grid and el_grid.
+
+        Parameters
+        ----------
+        total_time : `float`
+            Total observation time in seconds.
+        az_grid : `list` of `float`
+            Azimuth coordinates to slew and track.
+        el_grid : `list` of `float`
+            Elevation coordinates to slew and track.
+        el_cutoff : `float`
+            Elevation cutoff limit used to skip targets to minimize
+            the number of targets in high elevation.
+        """
+        step = 0
+        timer_task = asyncio.create_task(asyncio.sleep(total_time))
+        self.log.info(
+            f"{'Time':25s}{'Steps':>10s}{'Old Az':>10s}{'New Az':>10s}{'Old El':>10s}{'New El':>10s}"
+        )
+
+        generator = self.generate_azel_sequence(az_grid, el_grid, el_cutoff=el_cutoff)
+
+        n_points = 10
+        old_az = np.median(
+            [
+                self.tcs.rem.mtmount.tel_azimuth.get().actualPosition
+                for i in range(n_points)
+            ]
+        )
+        old_el = np.median(
+            [
+                self.tcs.rem.mtmount.tel_elevation.get().actualPosition
+                for i in range(n_points)
+            ]
+        )
+
+        while not timer_task.done():
+
+            new_az, new_el = next(generator)
+
+            if new_az == old_az and new_el == old_el:
+                continue
+
+            t = Time.now().to_value("isot")
+            self.log.info(
+                f"{t:25s}{step:10d}{old_az:10.2f}{new_az:10.2f}{old_el:10.2f}{new_el:10.2f}"
+            )
+
+            yield step, new_az, new_el
+            old_az, old_el = new_az, new_el
+            step += 1
+
+    def generate_azel_sequence(self, az_seq, el_seq, el_cutoff=90.0):
```

Comment:
So, one thing that I notice in this method is that it does not need to be a class member (e.g. it does not access any `self` value). Consider making this a static method, e.g.:

```
@staticmethod
def generate_azel_sequence(az_seq, el_seq, el_cutoff=90.0):
    ...
```